File: Two Furthest Houses With Different Colors/Two Furthest Houses With Different Colors.py
# Solution for Two Furthest Houses With Different Colors in PY

class Solution:
    def maxDistance(self, colors):
        total_houses = len(colors)
        maximum_distance = 0

        for current_index in range(total_houses):

            # Case 1: Compare with first house
            if colors[current_index] != colors[0]:
                maximum_distance = max(maximum_distance, current_index)

            # Case 2: Compare with last house
            if colors[current_index] != colors[total_houses - 1]:
                maximum_distance = max(maximum_distance, total_houses - 1 - current_index)

        return maximum_distance

File: Two Furthest Houses With Different Colors/Two Furthest Houses With Different Colors.java
// Solution for Two Furthest Houses With Different Colors in JAVA

class Solution {
    public int maxDistance(int[] colors) {
        int totalHouses = colors.length;
        int maximumDistance = 0;

        for (int currentIndex = 0; currentIndex < totalHouses; currentIndex++) {

            // Case 1: Compare with first house
            if (colors[currentIndex] != colors[0]) {
                maximumDistance = Math.max(maximumDistance, currentIndex);
            }

            // Case 2: Compare with last house
            if (colors[currentIndex] != colors[totalHouses - 1]) {
                maximumDistance = Math.max(maximumDistance, totalHouses - 1 - currentIndex);
            }
        }

        return maximumDistance;
    }
}
